library/src/main/java/com/google/android/exoplayer2/extractor/DefaultTrackOutput.java

```diff
@@ -229,10 +229,26 @@ public long getLargestQueuedTimestampUs() {
    * Attempts to skip to the keyframe before the specified time, if it's present in the buffer.
    *
    * @param timeUs The seek time.
+   * @param skipToLastKey Skip to last key regardless the seek time is out of range .
+   * @return Whether the skip was successful.
+   */
+  public boolean skipToKeyframeBefore(long timeUs, boolean skipToLastKey) {
+    long nextOffset = infoQueue.skipToKeyframeBefore(timeUs, skipToLastKey);
+    if (nextOffset == C.POSITION_UNSET) {
+      return false;
+    }
+    dropDownstreamTo(nextOffset);
+    return true;
+  }
+
+  /**
+   * Attempts to skip to the keyframe before the specified time, if it's present in the buffer.
+   *
+   * @param timeUs The seek time.
    * @return Whether the skip was successful.
    */
   public boolean skipToKeyframeBefore(long timeUs) {
-    long nextOffset = infoQueue.skipToKeyframeBefore(timeUs);
+    long nextOffset = infoQueue.skipToKeyframeBefore(timeUs, false);
```

Comment:
You can replace the whole body of this method with just:
`return skipToKeyframeBefore(timeUs, false)`